src/main/java/com/notably/logic/parser/NotablyParser.java

```diff
@@ -32,31 +41,31 @@
         if (!matcher.matches()) {
             throw new ParseException(String.format("Invalid Command"));
         }
-
-        final String commandWord = matcher.group("commandWord");
+        String commandWord = matcher.group("commandWord");
+        Optional<String> correctedCommand = correctionEngine.correct(commandWord).getCorrectedItem();
+        if (correctedCommand.equals(Optional.empty())) {
+            throw new ParseException("Invalid command");
+        }
+        commandWord = correctedCommand.get();
```

Comment:
Let's add a newline above and below this group to increase readability?

Reply:
Done,